### PR TITLE
Collecting secret and not-so-secret values

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Document how to collect cryptographically sensitive values negotiated in the connection

### Motivation and Context

Better documentation
depends on #653

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - n/a
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/654)

### CODE REVIEW

File: docs/source/statistical-analysis.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
+====================
+Statistical analysis
+====================
+
+As cryptographic security depens on proper use of the primitives, tests
```
Comment (sahanaprasad07) [inline docs/source/statistical-analysis.rst]: *depends

Comment (sahanaprasad07) [reply docs/source/statistical-analysis.rst]: *the proper use

File: docs/source/statistical-analysis.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+===============================
+
+Unlike nonces, negotiation or advertising of other cryptographic parameters
+happens just once per connection.
+To save those parameters use the
+:py:class:`~tlsfuzzer.messages.CopyVariables` node.
+For full list of supported parameters see the class documentation,
+you can find definitions of the names in the :term:`TLS` :term:`RFC`\ s.
+
+As parameter this node accepts a dictionary where keys specify names of
```
Comment (sahanaprasad07) [inline docs/source/statistical-analysis.rst]: *As a parameter, this node accepts a dictionary in which, keys specify the names of

File: docs/source/statistical-analysis.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+than one :py:class:`~tlsfuzzer.messages.CopyVariables`, it appends new
+values to the arrays, it doesn't replace the arrays.
+
+You can find a usage example of it in:
+`test-serverhello-random.py
+<https://github.com/tomato42/tlsfuzzer/blob/master/scripts/test-serverhello-random.py>`_.
+
+.. tip::
+
+    Tlsfuzzer provides a simple function to verify uniquness of parameters in
```
Comment (sahanaprasad07) [inline docs/source/statistical-analysis.rst]: *uniqueness

File: docs/source/statistical-analysis.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    such dictionary: :py:func:`~tlsfuzzer.helpers.uniqueness_check`.
```
Comment (sahanaprasad07) [inline docs/source/statistical-analysis.rst]: *such a dictionary?

File: docs/source/statistical-analysis.rst (hunk 1 above)
Comment (tomato42) [reply docs/source/statistical-analysis.rst]: updated

File: docs/source/statistical-analysis.rst (hunk 2 above)
Comment (tomato42) [reply docs/source/statistical-analysis.rst]: updated, but without commas, I don't think there're right here, especially the comma after "which"

File: docs/source/statistical-analysis.rst (hunk 3 above)
Comment (tomato42) [reply docs/source/statistical-analysis.rst]: done

File: docs/source/statistical-analysis.rst (hunk 4 above)
Comment (tomato42) [reply docs/source/statistical-analysis.rst]: done

**Pull request merged**